### PR TITLE
Add Split-18: new num_outputs attribute

### PR DESCRIPTION
**Pull request by p-wysocki**
Signed-off-by: p-wysocki <przemyslaw.wysocki@intel.com>

**Description**
- New optional Split attribute `num_outputs` determining the number of equally sized output tensors.

**Motivation and Context**
- Fixes an issue described at https://github.com/onnx/onnx/issues/4188

### CODE REVIEW

File: onnx/defs/tensor/defs.cc [hunk 1]
```diff
@@ -587,15 +587,15 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
-static const char* Split_ver13_doc =
-    R"DOC(Split a tensor into a list of tensors, along the specified
-'axis'. Lengths of the parts can be specified using input 'split'.
-Otherwise, the tensor is split to equal sized parts.
+static const char* Split_ver18_doc =
+    R"DOC(Either input 'split' or the attribute 'num_outputs' should be specified, but not both.
```
Comment (postrational) [inline onnx/defs/tensor/defs.cc:591]: This sentence should not be removed:

> Split a tensor into a list of tensors, along the specified 'axis'.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -31,13 +31,23 @@ from onnx root dir should work.
 
 ## Generated operator documentation
 
-[Operator docs in Operators.md](Operators.md) are automatically generated based on C++ operator definitions and backend Python snippets. To refresh these docs, run the following commands from the repo root and commit the results. Note `ONNX_ML=0` updates Operators.md whereas `ONNX_ML=1` updates Operators-ml.md:
+[Operator docs in Operators.md](Operators.md) are automatically generated based on C++ operator definitions and backend Python snippets. To refresh these docs, run the following commands from the repo root and commit the results.
```
Comment (postrational) [inline docs/CONTRIBUTING.md:34]: CONTRIBUTING.md should not change in this PR.

**Review by postrational — CHANGES_REQUESTED** [review]

**Comment by postrational** [comment]
Previously discussed in https://github.com/onnx/onnx/pull/4358

File: onnx/defs/tensor/defs.cc (hunk 1 above)
Comment (p-wysocki) [reply onnx/defs/tensor/defs.cc]: Good catch, I restored it.

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (p-wysocki) [reply docs/CONTRIBUTING.md]: I reverted the changes.

File: onnx/defs/tensor/defs.cc [hunk 3]
```diff
@@ -682,14 +688,21 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+                fail_shape_inference("Attribute `num_outputs` value cannot be lower than 1");
+              }
+              if (split_dim_value % num_outputs != 0) {
+                fail_shape_inference("The input is not evenly splittable");
+              }
+              int chunk_size = split_dim_value / num_outputs;
+              split.reserve(ctx.getNumOutputs());
+              for (int i = 0; i < static_cast<int>(ctx.getNumOutputs()); i++) {
+                split.push_back(chunk_size);
+              }
```
Comment (daquexian) [inline onnx/defs/tensor/defs.cc:703]: Suggestion:
```suggestion
              split.resize(ctx.getNumOutputs(), chunk_size);
```

File: onnx/backend/test/case/node/split.py [hunk 4]
```diff
@@ -17,7 +17,8 @@ def export_1d() -> None:
             'Split',
             inputs=['input'],
             outputs=['output_1', 'output_2', 'output_3'],
-            axis=0
+            axis=0,
+            num_outputs=3
```
Comment (gramalingam) [inline onnx/backend/test/case/node/split.py:21]: I think we should perhaps have test-cases for both the earlier version and newer version? @jcwchen ?

**Review by postrational — APPROVED** [review]

File: onnx/backend/test/case/node/split.py (hunk 4 above)
Comment (jcwchen) [reply onnx/backend/test/case/node/split.py:21]: After this PR: https://github.com/onnx/onnx/pull/3855, going forward all of node test cases for the same operator will be updated if there is an opset bump. If people would like to keep some old test cases for old opset versions, they need to specify the opset_version while creating node test cases. Take Loop op as an example, see this [line](https://github.com/onnx/onnx/blob/ea7c5204713cce87e8c22e5382833c68483f329b/onnx/backend/test/case/node/loop.py#L127).

Comment (p-wysocki) [reply onnx/backend/test/case/node/split.py:115]: I applied the changes you suggested. Opset13 tests were reverted, new (opset18) tests stayed the same.

**Comment by jcwchen** [comment]
For unknown reason (might be a GitHub server issue), this PR hasn't been merged into the main branch even after 1 hrs... Let's see whether it will be merged toady.

**Comment by gramalingam** [comment]
As per @p-wysocki , it is unclear why this (empty) merge even happened. He says he did nothing to trigger it. And it seems to be an empty merge with no commits. What's happening? Seems mysterious to me.

**Comment by p-wysocki** [comment]
I was 1 commit ahead of `onnx:main`, it contained all my changes, squashed with a sign off. Unfortunately some other commits got in and there were ~350 files changed. Long story short, I reverted this one, single commit from `p-wysocki:add_split18` and pushed it to my fork, essentially syncing `onnx:main` with `p-wysocki:add_split18`, what can be seen as "Files changed: 0". I think this may be just a weird behavior from GitHub, I did not enable merging (merge should not be possible at all anyway), my changes are not on master and it would be just a "visual" glitch. 

We had this happen in OpenVINO too, where an external contributor seemingly merged his changes, but they were not on master and the PR showed 0 changed files.

**Pull request merged**